test.js

```diff
@@ -158,6 +154,27 @@ describe('capture-exit', function() {
         exit.onExit(function () { });
       }).to.throw('Cannot install handler when exit is not captured.  Call `captureExit()` first');
     });
+
+    it('throws if an exit is already happening', function() {
```

Comment:
Because we can only handle this during our own `onExit` callbacks and not during `process.on('exit'` (since process.on('exit' cannot do any async).

Essentially, we are making the assumption that once we enter the "exit flow" that additional handlers cannot be added.